timemachine/fe/atom_mapping.py

```diff
@@ -187,11 +193,15 @@ def get_core_by_mcs(mol_a, mol_b, query, threshold=0.5):
     matches_a = mol_a.GetSubstructMatches(query, uniquify=False)
     matches_b = mol_b.GetSubstructMatches(query, uniquify=False)
 
-    # cost[i, j] = sum_i distance(conf)
     cost = np.zeros((len(matches_a), len(matches_b)))
     for i, a in enumerate(matches_a):
         for j, b in enumerate(matches_b):
-            cost[i, j] = np.linalg.norm(conf_a[np.array(a)] - conf_b[np.array(b)], axis=1).sum()
+            # if a single pair is outside of threshold, we set the cost to inf
+            dij = np.linalg.norm(conf_a[np.array(a)] - conf_b[np.array(b)], axis=1)
+            if np.any(dij > threshold):
+                cost[i, j] = np.inf
+            else:
+                cost[i, j] = dij.sum()
```

Comment:
nit: slightly cleaner as ternary expression
```suggestion
            cost[i, j] = dij.sum() if np.all(dij < threshold) else np.inf
```